app/src/main/java/com/flowfoundation/wallet/page/browser/widgets/LilicoWebView.kt

```diff
@@ -270,10 +293,28 @@ class LilicoWebView : WebView {
                         e.printStackTrace()
                         false
                     }
+                } else if (it.scheme == "tg") {
+                    val openTg = Intent(Intent.ACTION_VIEW, it)
+                        .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+
+                    try {
+                        context.startActivity(openTg)
+                    } catch (e: Exception) {
+                        Toast.makeText(context, R.string.telegram_not_installed, Toast.LENGTH_SHORT).show()
+                    }
+
+                    // Stop the WebView navigation to avoid looping
+                    view?.stopLoading()
+                    view?.clearHistory()
+                    return true
                 } else if (it.host == "link.lilico.app" || it.host == "frw-link.lilico.app" || it
```

Comment:
let's register all universal link as a enum class, then check the host in the list of enum.
```
enum UniversaLinks: String {
case frw = ""
case fcw = ""
case wallet = ""
}

enum Deeplinks: {
case frw
case fcw
}

UniversaLinks.values.contains(host)
Deeplinks.values.contains(scheme)
```